chrome/content/zotero/xpcom/data/item.js

```diff
@@ -732,6 +732,10 @@ Zotero.Item.prototype.setField = function(field, value, loadIn) {
 		}
 		return true;
 	}
+
+	if (!loadIn && field == 'language') {
+		value = Zotero.Utilities.Item.languageToISO6391(value);
+	}
```

Comment:
This should go down below, within the existing `if (!loadIn) {` block on line 805. (This way, I don't think a non-trimmed value would even work.)